core/workspace_svg.js

```diff
@@ -1348,6 +1348,15 @@ Blockly.WorkspaceSvg.prototype.zoomCenter = function(type) {
   var metrics = this.getMetrics();
   var x = metrics.viewWidth / 2;
   var y = metrics.viewHeight / 2;
+  if (this.options.toolboxPosition == 2) {
```

Comment:
Do you mean the metric.viewHeight or the toolbox Position, if the toolbox position, is there any other way to detect it?